lib/Sema/CSSimplify.cpp

```diff
@@ -2972,6 +2972,11 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       
     } else {
       if (!hasStaticMembers) {
+        if (cand->isImplicit() && isa<ConstructorDecl>(cand)) {
```

Comment:
It might be too late at that point. I don't think we can *remove* a declaration. But we can mark the declaration invalid, and the verifier will assert that someone else has emitted some error.